File: typescript/ex_005_resistor-color-trio/resistor-color-trio.ts
const COLORS: string[] = 'black brown red orange yellow green blue violet grey white'.split(' ');
const MAGNITUDES: string[] = ',kilo,mega,giga'.split(',');

export function decodedResistorValue(input: string[]): string {
  const [n1, n2, n3] = input;
  let result: number = (COLORS.indexOf(n1) * 10 + COLORS.indexOf(n2)) * 10 ** COLORS.indexOf(n3);
  let count: number = 0;
  while(result && result % 1000 === 0) {
    count++;
    result /= 1000;
  }
  return result + ' ' + MAGNITUDES[count] + 'ohms';
}

File: java/ex_034_resistor-color-trio/ResistorColorTrio.java
import java.util.Arrays;

class ResistorColorTrio {
    String label(String[] colors) {
        long result = value(colors);
        int count = 0;
        while(result != 0 && result % 1000 == 0) {
          count++;
          result /= 1000;
        }
        return result + " " + magnitudes(count) + "ohms";
    }

    long value(String[] colors) {
        return (colorCode(colors[0]) * 10 + colorCode(colors[1])) * (long)Math.pow(10, colorCode(colors[2]));
    }
    
    int colorCode(String color) {
        return Arrays.asList(colors()).indexOf(color);
    }
    
    String[] colors() {
        return "black,brown,red,orange,yellow,green,blue,violet,grey,white".split(",");
    }

    String magnitudes(int index) {
        return ",kilo,mega,giga".split(",")[index];
    }
}
